contracts/Nexus.sol

```diff
@@ -72,22 +80,25 @@ contract Nexus is INexus, EIP712, BaseAccount, ExecutionHelper, ModuleManager, U
     ///     - `SIG_VALIDATION_FAILED` (1) denotes signature validation failure allowing simulation calls without a valid signature.
     /// @dev Expects the validator's address to be encoded in the upper 96 bits of the userOp's nonce.
     /// This method forwards the validation task to the extracted validator module address.
+    /// @dev Features Module Enable Mode.
+    /// This Module Enable Mode flow only makes sense for the module that is used as validator
+    /// for the userOp that triggers Module Enable Flow. Otherwise, one should just include
+    /// a call to Nexus.installModule into userOp.callData
     function validateUserOp(
-        PackedUserOperation calldata userOp,
+        PackedUserOperation calldata op,
         bytes32 userOpHash,
         uint256 missingAccountFunds
     ) external virtual payPrefund(missingAccountFunds) onlyEntryPoint returns (uint256 validationData) {
-        address validator;
-        uint256 userOpnonce = userOp.nonce;
-        assembly {
-            validator := shr(96, userOpnonce)
-        }
-        // Check if validator is not enabled. If not, return VALIDATION_FAILED.
-
-        if (!_isValidatorInstalled(validator)) return VALIDATION_FAILED;
-
-        // bubble up the return value of the validator module
-        validationData = IValidator(validator).validateUserOp(userOp, userOpHash);
+        address validator = op.nonce.getValidator();
+        if (op.nonce.isModuleEnableMode()) {
```

Comment:
Which condition do you think is more likely to happen, this one or the other?

Reply:
idk. I hope that the permissions won't be one-timers. so switched